### ISSUE TITLE
Problem in /extension/history.js ?

### ISSUE THREAD

**Issue by headwarsun** [SupportClass/lfg-sublistener#5]
Hello,

In `history.js` line 37:

``` javascript
if (hist[channel]) {
```

always returns `false` the first time it is called, as `hist[channel]` has not yet been initialized. It will be at the first `.add()` call (lines 14-16), which should always happen as `.exists()` returned `false`.

As a consequence, each time NodeCG launches, seeing that `hist.exists()` is always called before `hist.add()`, one false 'subscription' event is triggered and one duplicate entry is written in the `./db/lfg-sublistener` database.

A simple fix, I think, would be to initialize `hist[channel]` in the `.exists()` method, ie replace line 37 by:

``` javascript
  if (!hist.hasOwnProperty(channel)) {
            hist[channel] = jsonStorage.getItem(channel) || [];
        }        
        if (hist[channel].length > 0) {
```

**Comment by Lange** [comment]
You could very well be right!

PR welcome.

**Comment by Lange** [comment]
Fixed by #6
